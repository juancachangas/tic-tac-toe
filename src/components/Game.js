/*
 * You do not need to separate components and containers.
 * Feel free to connect your state and components directly in here (or other components).
 */
import React from 'react';
import { connect } from 'react-redux';

import Board from './Board';
import Modal from './Modal';
import { restartBoard } from '../store/actions/board';

export const Game = ({ status, restart }) => (
  <div>
    <header>
      TicTacToe
      <button type="button" onClick={() => restart()}> New Game</button>
    </header>

    <div className="Game">
      <Board />
    </div>
    {
      !['playing', 'new'].includes(status)
      && (
        <Modal status={status} names={['juan', 'camilo']} />
      )
    }
  </div>
);
export const mapStateToProps = (state) => {
  const { status, turn } = state.board;
  return {
    status,
    turn,
  };
};
export const mapDispatchToProps = dispatch => ({
  restart: () => dispatch(restartBoard()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Game);
