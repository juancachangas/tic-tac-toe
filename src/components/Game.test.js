import React from 'react';
import { shallow } from 'enzyme';

import Game from './Game';

describe('Game component is', () => {
  it('rendering default', () => {
    const wrapper = shallow(
      <Game />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
