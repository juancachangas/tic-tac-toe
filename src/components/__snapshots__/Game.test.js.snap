// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component is rendering default 1`] = `
<div>
  <header>
    TicTacToe
    <button
      onClick={[Function]}
      type="button"
    >
       New Game
    </button>
  </header>
  <div
    className="Game"
  >
    <Board />
  </div>
</div>
`;

exports[`Game component is rendering draw 1`] = `
<div>
  <header>
    TicTacToe
    <button
      onClick={[Function]}
      type="button"
    >
       New Game
    </button>
  </header>
  <div
    className="Game"
  >
    <Board />
  </div>
  <Modal
    names={
      Array [
        "juan",
        "camilo",
      ]
    }
    status="draw"
  />
</div>
`;

exports[`Game component is rendering error 1`] = `
<div>
  <header>
    TicTacToe
    <button
      onClick={[Function]}
      type="button"
    >
       New Game
    </button>
  </header>
  <div
    className="Game"
  >
    <Board />
  </div>
  <Modal
    names={
      Array [
        "juan",
        "camilo",
      ]
    }
    status="wrong_move"
  />
</div>
`;

exports[`Game component is rendering winner-0 1`] = `
<div>
  <header>
    TicTacToe
    <button
      onClick={[Function]}
      type="button"
    >
       New Game
    </button>
  </header>
  <div
    className="Game"
  >
    <Board />
  </div>
  <Modal
    names={
      Array [
        "juan",
        "camilo",
      ]
    }
    status="winner-0"
  />
</div>
`;

exports[`Game component is rendering winner-1 1`] = `
<div>
  <header>
    TicTacToe
    <button
      onClick={[Function]}
      type="button"
    >
       New Game
    </button>
  </header>
  <div
    className="Game"
  >
    <Board />
  </div>
  <Modal
    names={
      Array [
        "juan",
        "camilo",
      ]
    }
    status="winner-1"
  />
</div>
`;
